src/browser.js

```diff
@@ -53,9 +55,9 @@ export default function cssHasPseudo(document) {
 							nodes.push(element);
 
 							// set an attribute with an irregular attribute name
-							attributeElement.innerHTML = '<x ' + item.attributeName + '>';
+							attributeElement.outerHTML = '<x ' + item.attributeName + '>';
 
-							element.attributes.setNamedItem(attributeElement.firstChild.attributes[0].cloneNode());
+							element.setAttributeNode(attributeElement.attributes[0].cloneNode());
```

Comment:
`setAttributeNode` is a great idea. In fact, do you think this would also work:

```suggestion
element.setAttributeNode(attributeElement.getAttributeNode(item.attributeName));
```

Reply:
Sorry what? Your suggested `children[0].getAttributeNode(item.attributeName)` is a lot longer than `firstChild.attributes[0]`

Reply:
Yes, but `firstChild` and `attributes` are 2 new expressions, so replacing them with existing expressions increases compression size.

Reply:
You are right. It seems the absolute most efficient is:

```suggestion
							element.setAttributeNode(attributeElement.firstChild.attributes[0].cloneNode());
```

Reply:
It should have been `children[0].attributes[0]`, which is the most efficient for gzip compression.